### DISCUSSION TITLE
Running on Docker

### DISCUSSION THREAD

**Discussion by Konuralpkilinc** [huggingface/chat-ui#1331] [Q&A]
The computer that i will be using to run the chat-ui lacks internet connection and also authorization (and also Docker). Can I use the following codes "docker run -p 3000:3000 --env-file .env.local --name chat-ui ghcr.io/huggingface/chat-ui" by using podman if I already have the chat-ui.tar for dependency (I presume I can run "docker run -d -p 27017:27017 -v mongo-chat-ui:/data --name mongo-chat-ui mongo:latest" in podman). Thanks for any help in advance.

**Comment by Konuralpkilinc** [comment]
I started the mongodb by "docker run -d -p 27017:27017 mongotar_image_id" and  "docker run -p 3000:3000 --env-file .env.local --name chat-ui ghcr.io/huggingface/chat-ui" and set the .env.local by  

MONGODB_URL=mongodb://remote_server_ip:27017/
MODELS=`[{"name":"codegemma-2b-IQ1_S.gguf","preprompt":"","chatPromptTemplate":"<s>{{preprompt}}{{#each messages}}{{#ifUser}}\n{{content}}\n\n{{/ifUser}}{{#ifAssistant}}{{content}}\n{{/ifAssistant}}{{/each}}","parameters":{"stop":["","",""],"temperature":0.7,"max_new_tokens":1024,"truncate":3071},"endpoints":[{"type":"llamacpp","baseURL":"http://localhost:8080"}]}]`

However i can't manage to navigate to the remote_server_ip:3000 it gives error code of 500 and the docker logs are like this :

{"level":30,"time":1720630320826,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] Begin check..."}
{"level":30,"time":1720630320843,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] \"Update search assistants\" already applied. Skipping..."}
{"level":30,"time":1720630320843,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] \"Update deprecated models in assistants with the default model\" should not be applied for this run. Skipping..."}
{"level":30,"time":1720630320844,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] \"Add empty 'tools' record in settings\" already applied. Skipping..."}
{"level":30,"time":1720630320844,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] \"Convert message updates to the new schema\" already applied. Skipping..."}
{"level":30,"time":1720630320844,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] \"Convert message files to the new schema\" already applied. Skipping..."}
{"level":30,"time":1720630320844,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] \"Trim message updates to reduce stored size\" already applied. Skipping..."}
{"level":30,"time":1720630320844,"pid":22,"hostname":"3d158a8b4d5f","msg":"[MIGRATIONS] All migrations applied. Releasing lock"}
Listening on 0.0.0.0:3000
{"level":30,"time":1720630320871,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"day","type":"conversation","msg":"Computing conversation stats"}
{"level":30,"time":1720630320872,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"updatedAt","span":"day","type":"conversation","msg":"Computing conversation stats"}
{"level":30,"time":1720630320875,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"day","type":"message","msg":"Computing conversation stats"}
{"level":30,"time":1720630320876,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"updatedAt","span":"week","type":"conversation","msg":"Computing conversation stats"}
{"level":30,"time":1720630320876,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"week","type":"conversation","msg":"Computing conversation stats"}
{"level":30,"time":1720630320877,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"week","type":"message","msg":"Computing conversation stats"}
{"level":30,"time":1720630320878,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"updatedAt","span":"month","type":"conversation","msg":"Computing conversation stats"}
{"level":30,"time":1720630320879,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"month","type":"conversation","msg":"Computing conversation stats"}
{"level":30,"time":1720630320880,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"day","type":"message","msg":"Computed conversation stats"}
{"level":30,"time":1720630320880,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"updatedAt","span":"week","type":"conversation","msg":"Computed conversation stats"}
{"level":30,"time":1720630320880,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"day","type":"conversation","msg":"Computed conversation stats"}
{"level":30,"time":1720630320881,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"updatedAt","span":"day","type":"conversation","msg":"Computed conversation stats"}
{"level":30,"time":1720630320881,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"week","type":"conversation","msg":"Computed conversation stats"}
{"level":30,"time":1720630320882,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"week","type":"message","msg":"Computed conversation stats"}
{"level":30,"time":1720630320882,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"updatedAt","span":"month","type":"conversation","msg":"Computed conversation stats"}
{"level":30,"time":1720630320882,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"month","type":"message","msg":"Computing conversation stats"}
{"level":30,"time":1720630320883,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"month","type":"conversation","msg":"Computed conversation stats"}
{"level":30,"time":1720630320885,"pid":22,"hostname":"3d158a8b4d5f","minDate":"1970-01-01T00:00:00.000Z","dateField":"createdAt","span":"month","type":"message","msg":"Computed conversation stats"}
{"level":50,"time":1720630736403,"pid":22,"hostname":"3d158a8b4d5f","locals":{"sessionId":"e1af388894d4d1b09003ac5e6b1fb6ffb9607408ef5b55b016ebc457d8fd17b1"},"url":"https://remote_server_ip:3000/","params":{},"request":{},"error":{"lineNumber":1,"columnNumber":1},"errorId":"691d90f4-bf87-4228-89d9-18604899eca9"}

Where am I making the mistake?

**Reply by sebastiangonsal** [reply]
were you able to fix this ? I'm hitting the same issue
